File: FactorySimulation/src/factory/Factory.java
package factory;

import java.util.List;

public class Factory {
	
	List<FactoryLine> factoryLines;

	public List<FactoryLine> getFactoryLines() {
		return factoryLines;
	}

	public void setFactoryLines(List<FactoryLine> factoryLines) {
		this.factoryLines = factoryLines;
	}
	
	
	public boolean isFactoryEmpty() {
		for(FactoryLine l: factoryLines) {
			for(Unit u  : l.getUnits()) {
				for(Machine m : u.getMachines()) {
					if(m.getWorkingOn() >0) {
						System.out.println("Factory not empty");
						return false;
					}
				}
			}
		}
		
		
		return true;
	}
	
	

}

File: FactorySimulation/C# Simulation/FactorySimulation/Factory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FactorySimulation
{
    class Factory
    {

		private List<Line> _factoryLines;

		public List<Line> GetFactoryLines()
		{
			return _factoryLines;
		}

		public void SetFactoryLines(List<Line> factoryLines)
		{
			this._factoryLines = factoryLines;
		}


		public bool IsFactoryEmpty()
		{
			foreach (Line l in _factoryLines)
			{
				foreach (Unit u  in l.GetUnits())
				{
					foreach (Machine m in u.GetMachines())
					{
						if (m.GetWorkingOn() > 0)
						{

							return false;
						}
					}
				}
			}


			return true;
		}


	}
}
